### PR TITLE
Add support for importing inventory and products

### PR DESCRIPTION
**Pull request by Wambere**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Part of #157 

Built this branch off of #178 so it might look a bit more loaded 

**Engineer Checklist**
- [ ] I have run `./gradlew spotlessApply` to check my code follows the project's style guide
- [ ] I have built and run the efsity jar to verify my change fixes the issue and/or does not break the application

### CODE REVIEW

**Comment by Wambere** [comment]
@dubdabasoduba please review the resources created for products and inventories here

@pld @ukanga I'm going to use the other PR ( #178 ) for a proper refactor, just trying to unblock this here

File: importer/main.py [hunk 1]
```diff
@@ -475,9 +525,228 @@ def care_team_extras(
...
+
+        if donor:
+            payload_obj["resource"]["characteristic"][2]["valueCodeableConcept"][
+                "text"
+            ] = donor
+        else:
+            del_indexes.append(GROUP_INDEX_MAPPING["inventory_donor_index"])
+
+        for x in reversed(del_indexes):
+            del payload_obj["resource"]["characteristic"][x]
```
Comment (pld) [inline importer/main.py:736]: think you can move this after the conditional, like to line 740

File: importer/main.py [hunk 2]
```diff
@@ -475,9 +525,228 @@ def care_team_extras(
...
+            else:
+                logging.error(
+                    "Unable to link the image Binary resource for product " + item_name
+                )
+                del_indexes.append(GROUP_INDEX_MAPPING["product_image_index"])
+        else:
+            del_indexes.append(GROUP_INDEX_MAPPING["product_image_index"])
+
+        for x in reversed(del_indexes):
+            del payload_obj["resource"]["characteristic"][x]
```
Comment (pld) [inline importer/main.py:631]: can remove entirely if you move this or other one below the conditional

**Review by pld — COMMENTED** [review]
this looks pretty good, we could probably make it a little tighter but we'd have to get into python internals in a way that would make it less readable, open to ideas, but I'd be comfortable merging this in

**Review by pld — APPROVED** [review]

**Pull request merged**